distributed/protocol/compression.py

```diff
@@ -93,6 +93,27 @@ def _fixed_lz4_decompress(data):
     }
     default_compression = "lz4"
 
+
+with ignoring(ImportError):
+    import zstandard as zstd
+
+    def zstd_compress(data):
+        if isinstance(data, (memoryview, bytearray)):
+            data = bytes(data)
+        return zstd.ZstdCompressor().compress(data)
+
+    def zstd_decompress(data):
+        if isinstance(data, (memoryview, bytearray)):
+            data = bytes(data)
+        return zstd.ZstdDecompressor().decompress(data)
+
+    compressions["zstd"] = {
+        "compress": zstd_compress,
+        "decompress": zstd_decompress,
+    }
+    default_compression = "zstd"
```

Comment:
Hrm, should zstd or lz4 be the default?

Reply:
that's a good question. The [squash benchmark](https://quixdb.github.io/squash-benchmark/) kinda indicates that zstd is better, but that of course depends on the payload.

Reply:
My previous intuition was that lz4 ran at a few GB/s bandwidth and got modest compression ratios (2-4) while zstd ran a bit slower, maybe 1GB/s, but got better compression ratios (4-10).  Obviously those numbers are highly dependent.

Reply:
The question is though should `zstd` be made the default over lz4. I think we'd want to benchmark a bit on realistic workloads before making this change for users. Not all users set the configuration system, the default should be the best option and I'm not yet convinced zstd is better for our workloads.

Reply:
@jcrist  @mrocklin  @crepererum @jbcrail 
vote !    zstd +1 

Dask Congress count the tickets.

zstd, lz4, snappy, ...   all were selected by people in a long time.
reference:   
    https://facebook.github.io/zstd/
    https://pypi.org/project/zstandard/
    .....

----------------------------------------------------------------
Zstandard is used by :

Featured

 Linux    FreeBSD   Redshift    Mercurial
Databases

 RocksDB    Hadoop    Cassandra    WiredTiger    Redis    Presto    Clickhouse    Groonga  Tarantool    TokuDB    HBase    PostgreSQL    VictoriaMetrics
File systems

 BtrFS    SquashFS    OpenZFS    ReiserFS
Web

 caddy    HHVM    nginx (module)    wget2    django
Serialization

 FST    Blosc    bcolz    mrcz    bgen    Gecko    HDF5    Sereal    PLink2    NAF
Network

 fbthrift    Fizz    proxygen    mcrouter    Rspamd    Tor
Hardware

 Inaccel
Games

 League of Legends    Godot    Khronos's KTX    OpenSiv3D    Esenthel    Mass Effect:Andromeda
Other

 Ubuntu    IBM TS7700    folly    cmake    TaskCluster   libarchive    tar    PowerAchiver  Fog    Backup    U++    Ceph   LiveScan3D